tests/test_parser.py

```diff
@@ -1103,3 +1114,52 @@ def test_parameter_braces(self):
               ])
             ),
         )
+
+    def test_cases(self):
+        #return
+        s = """
+            case "$1" in
+            1) echo foo
+            ;;
+            *)
+            (
+                echo bar
+            )
+            ;;
+            esac
+            """
+        self.assertASTEquals(s,
```

Comment:
i see what you mean about the resulting ast :) is there a more condensed way to represent it, that is still in line with how everything else is parsed?